pytorch_lightning/trainer/trainer.py

```diff
@@ -456,7 +456,9 @@ def fit(
         self.call_hook("on_before_accelerator_backend_setup", model)
         self.accelerator.connect(model)
         self.accelerator.setup_environment()
+        self.accelerator.barrier("pre-setup")
         self.call_setup_hook(model)  # allow user to setup lightning_module in accelerator environment
+        self.accelerator.barrier("post-setup")
```

Comment:
can the barrier be moved into the "call_setup_hook"?

Reply:
it could but it's a style choice - should the hooks handle the synchronization, or should it be the trainer/loop?